### PR TITLE
BUILDS: Use vendor directory instead of Godeps

### PR DESCRIPTION
**Pull request by tomdee**
Closes #407

### CODE REVIEW

**Comment by tomdee** [comment]
ping @philips @steveeJ Any objections to this going in?

**Comment by tomdee** [comment]
Kind of blocked on #407 which is blocked on someone finding the time to do some GCE testing.

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,6 @@
 language: go
 go:
-  - 1.4
-  - 1.5
```
Comment (steveeJ) [inline .travis.yml]: should probably be testing with 1.5

Comment (steveeJ) [inline .travis.yml]: and then add `GO15VENDOREXPERIMENT=1` as an environment variable

**Comment by tomdee** [comment]
OK, this is updated to move to glide. This was _much_ easier for me to get working than Godep. (See https://github.com/coreos/docs/issues/775 for general CoreOS support for migrating to glide)

This fairly large PR also updates some upstream deps which have been removed (the Google code ones) and update vishvanda/netlink

It updates the travis.yaml file to make it very similar to the CNI one. The crucial change is removal of Go 1.4 support (since a vendor directory is now being used)

@steveeJ / @philips Can I get a LGTM so I can get this merged?

**Comment by tomdee** [comment]
I've verified that this fixes the vxlan port number problem.

**Pull request merged**